data-prepper-logstash-configuration/src/test/java/org/opensearch/dataprepper/logstash/mapping/OpenSearchPluginAttributesMapperTest.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.logstash.mapping;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasKey;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.opensearch.dataprepper.logstash.mapping.OpenSearchPluginAttributesMapper.LOGSTASH_OPENSEARCH_INDEX_ATTRIBUTE_NAME;
+
+import org.junit.jupiter.api.Test;
+import org.opensearch.dataprepper.logstash.model.LogstashAttribute;
+import org.opensearch.dataprepper.logstash.model.LogstashAttributeValue;
+import org.opensearch.dataprepper.logstash.model.LogstashValueType;
+
+import java.util.Collections;
+import java.util.Map;
+
+class OpenSearchPluginAttributesMapperTest {
+
+    private OpenSearchPluginAttributesMapper createObjectUnderTest() {
+        return new OpenSearchPluginAttributesMapper();
+    }
+
+    @Test
+    void convert_logstash_index_date_time_pattern() {
+        final String dataPrepperIndexAttribute = "index";
+        final String value = "my-application-index";
+
+        final LogstashAttribute logstashAttribute = mock(LogstashAttribute.class);
+        final LogstashAttributeValue logstashAttributeValue = mock(LogstashAttributeValue.class);
+        when(logstashAttributeValue.getValue()).thenReturn(value);
+        when(logstashAttribute.getAttributeName()).thenReturn("index");
+        when(logstashAttribute.getAttributeValue()).thenReturn(logstashAttributeValue);
+
+        final LogstashAttributesMappings logstashAttributesMappings = mock(LogstashAttributesMappings.class);
+        when(logstashAttributesMappings.getMappedAttributeNames()).thenReturn(Collections.singletonMap(LOGSTASH_OPENSEARCH_INDEX_ATTRIBUTE_NAME, dataPrepperIndexAttribute));
+
+
+        final Map<String, Object> pluginSettings =
+                createObjectUnderTest().mapAttributes(Collections.singletonList(logstashAttribute), logstashAttributesMappings);
+
+        assertThat(pluginSettings, notNullValue());
+        assertThat(pluginSettings.size(), equalTo(1));
+        assertThat(pluginSettings, hasKey(dataPrepperIndexAttribute));
+    }
+
+    @Test
+    void convert_logstashIndexPattern_joda_year_to_dataPrepperIndexPattern_java_year() {
```

Comment:
I think that this test and the next three are the same except for the input and output. You can make this a JUnit 5 parameterized test and this would be clearer to use.

Here is somewhat of a starting point for it.

```
    static class JodaToJava8IndicesArgumentsProvider implements ArgumentsProvider {
        @Override
        public Stream<? extends Arguments> provideArguments(final ExtensionContext extensionContext) {
            return Stream.of(
                    Arguments.arguments("logstash-%{+yyyy.MM.dd}", "logstash-%{uuuu.MM.dd}"),
                    Arguments.arguments("logstash-index-%{+YYYY.MM.dd}", "logstash-index-%{yyyy.MM.dd}"),
                    // repeat for each
            );
        }
    }
```


You can see this in use at:

https://github.com/opensearch-project/data-prepper/blob/main/data-prepper-logstash-configuration/src/test/java/org/opensearch/dataprepper/logstash/LogstashConfigConverterIT.java#L52-L54